File: 0x08-python-more_classes/1-rectangle.py
#!/usr/bin/python3

""" a square module"""

class Rectangle:
    """this defines a rectangle class"""
    def __init__(self, width=0, height=0):
        """This is a class constructor"""
        if isinstance(width, int):
            raise TypeError("width must be an integer")
        if width < 0:
            raise ValueError("width must be >= 0")
        self.__width = width
        if isinstance(height, int):
            raise TypeError("height must be an integer")
        if heigth < 0:
            raise ValueError("height must be >= 0")
        self.__height = height
    

File: 0x13-javascript_objects_scopes_closures/1-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    this.width = w;
    this.height = h;
  }
  print () {
    let i;
    let j;
    let row = '';
    for (i = 0; i < this.width; i++) {
      row += 'X';
    }
    for (j = 0; j < this.height; j++) {
      console.log(row);
    }
  }
}
module.exports = Rectangle;
